apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -1149,6 +1154,36 @@ private void HandleAckResponse(Task writeTask, List<string> ackIds, List<string>
                 // Immediately send more data if there is any to send.
                 StartPush();
             }
+
+            private void HandleStreamPing()
+            {
+                // Need to explicitly check this, as the continuation passed to Add() may be executed
+                // regardless of the fault/cancellation state of the Task.
+                if (_softStopCts.IsCancellationRequested)
+                {
+                    // No more pings when subscriber stopping.
+                    return;
+                }
+                // Schedule next ping, this never stops whilst this subscriber as active
+                Add(_scheduler.Delay(s_streamPingPeriod, _softStopCts.Token), Next(false, HandleStreamPing));
+                // If messages are currently being processed, then ping the stream periodically;
+                // this ensures the stream isn't closed.
+                // If the stream is closed, then all gRPC-buffered messages have their server-side
+                // expiry timers started, when the client hasn't yet started processing these messages.
+                // This can lead to unncessarily duplicated messages.
+                if (!IsPushComplete() && _pull != null)
+                {
+                    // Write an empty message to the stream
+                    try
+                    {
+                        _pull.WriteAsync(new StreamingPullRequest());
```

Comment:
We're deliberately not waiting for this to do anything, right? Might be worth a comment.